src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/resource/ResidentialChecksResource.kt

```diff
@@ -135,4 +142,62 @@ class ResidentialChecksResource(private val residentialChecksService: Residentia
     @Parameter(required = true) @PathVariable taskCode: String,
   ) =
     residentialChecksService.getResidentialChecksTask(prisonNumber, requestId, taskCode)
+
+  @PostMapping("/offender/{prisonNumber}/current-assessment/address-request/{requestId}/residential-checks/answers")
+  @PreAuthorize("hasAnyRole('ASSESS_FOR_EARLY_RELEASE_ADMIN')")
+  @ResponseStatus(code = HttpStatus.CREATED)
+  @Operation(
+    summary = "Saves answers for a residential checks task.",
+    description = "Save the answers for a residential checks task.",
+    security = [SecurityRequirement(name = "assess-for-early-release-admin-role")],
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "201",
+        description = "The task answers have been saved.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ResidentialChecksTaskAnswersSummary::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "An address check request with the specified request id does not exist for the provided offender",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun <T> saveResidentialChecksTaskAnswers(
+    @Parameter(required = true) @PathVariable prisonNumber: String,
+    @Parameter(required = true) @PathVariable requestId: Long,
+    @Valid @RequestBody taskAnswers: SaveResidentialChecksTaskAnswersRequest,
+  ): ResidentialChecksTaskAnswersSummary = residentialChecksService.saveResidentialChecksTaskAnswers(prisonNumber, requestId, taskAnswers)
```

Comment:
We'll probably want some state transition stuff here to transition to `ADDRESS_AND_RISK_CHECKS_IN_PROGRESS`, but doesn't need to be part of this PR!